### PR TITLE
Help: Hide the courses button on the help page for non business users

### PR DESCRIPTION
**Pull request by omarjackman**
This pull request hides the courses button for users without the business plan.

### How to test this
1. Navigate to https://calypso.live/help?branch=update/help-courses-hide-for-non-business-users
2. Notice the blinking placeholder.
3. If you are a business plan user the "Courses" button will appear. In other cases the placeholder will just disappear.

**Before**
screen shot 2016-08-24 at 8 01 49 pm

**After**
screen shot 2016-08-24 at 8 02 00 pm

Test live: https://calypso.live/?branch=update/help-courses-hide-for-non-business-users

### CODE REVIEW

**Comment by omarjackman** [comment]
cc @dllh @jordwest

File: client/me/help/main.jsx [hunk 1]
```diff
@@ -127,29 +151,39 @@ const Help = React.createClass( {
...
-			showCoursesTeaser
+			userId
 		} = this.props;
 		return (
 			<Main className="help">
 				<MeSidebarNavigation />
 				<HelpSearch />
 				{ ! isEmailVerified && <HelpUnverifiedWarning /> }
-				{ showCoursesTeaser && this.getCoursesTeaser() }
+				{ this.getCoursesTeaser() }
```
Comment (jordwest) [inline client/me/help/main.jsx]: Why the change here? Previously, if `showCoursesTeaser` was `false` the loading placeholder wouldn't show, which seems a bit less jarring if it doesn't eventually load something?

File: client/me/help/main.jsx [hunk 2]
```diff
@@ -127,29 +151,39 @@ const Help = React.createClass( {
...
 
-module.exports = reactRedux.connect(
+export default connect(
 	( state, ownProps ) => {
-		const isEmailVerified = currentUser.isCurrentUserEmailVerified( state );
-		const showCoursesTeaser = ownProps.isCoursesEnabled;
+		const isEmailVerified = isCurrentUserEmailVerified( state );
+		const userId = getCurrentUserId( state );
+		const purchases = getUserPurchases( state, userId );
+		const isLoading = ! hasLoadedUserPurchasesFromServer( state );
```
Comment (jordwest) [inline client/me/help/main.jsx]: Perhaps `isFetchingUserPurchases` would be a better fit here?

It's unlikely, but if loading fails I have a feeling that the placeholder would keep flashing forever, while `isFetchingUserPurchases` should stop flashing if it gives up.

**Comment by jordwest** [comment]
I've left a couple of comments, but mainly I'm wondering if a loading placeholder is the best idea here. Since the section is only supposed to show for business users, other users might be confused by the loading placeholder that suddenly disappears:

loading

Maybe @mrheston can weigh in here? Otherwise, the code looks good, so if there aren't any concerns with that let's 🚢

**Comment by mrheston** [comment]
I agree with @jordwest, having the placeholder and suddenly hiding it is confusing because it feels like something is broken or not loading properly. 

Ideally this should load (or not) with the rest of the elements in the page, but if that's not possible, I'd rather make it appear suddenly when needed.

**Comment by omarjackman** [comment]
@mrheston How do you feel about adding placeholders for all of the content on the page then. That probably would have been the best thing to do.

File: client/me/help/main.jsx (hunk 1 above)
Comment (omarjackman) [inline client/me/help/main.jsx]: Now place holds show for all of the cards

**Comment by mrheston** [comment]
@omarjackman But if the courses button doesn't show up in its placeholder, the rest of the content will jump upwards anyway, right? I'm not sure that will make it much better, but we can definitely try it.

File: client/me/help/main.jsx (hunk 2 above)
Comment (omarjackman) [inline client/me/help/main.jsx]: I swapped it out in my last rebase and in 313acb6

**Comment by omarjackman** [comment]
@mrheston I just rebased and added all the place holders, take another look :)

**Comment by mrheston** [comment]
This works much better! The actual content doesn't jump, so it's not confusing as it was before 👍
